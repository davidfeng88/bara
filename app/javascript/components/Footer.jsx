import React from 'react';
import PropTypes from 'prop-types';

const Footer = () => (
  <div className="footer">
    <div className="container footer-container">
      <FooterLinks />
      <FooterPicture />
    </div>
  </div>
);

export default Footer;

const FooterLinks = () => (
  <div className="footer-links">
    <AboutBara />
    <AboutMe />
    <Credits />
  </div>
);

const AboutBara = () => (
  <FooterLinkColumnTemplate
    title="About Bara"
  >
    <GitHubRepo />
  </FooterLinkColumnTemplate>
);

const AboutMe = () => (
  <FooterLinkColumnTemplate
    title="About Me"
  >
    <PortfolioSite />
    <GitHub />
    <LinkedIn />
    <AngelList />
  </FooterLinkColumnTemplate>
);

const Credits = () => (
  <FooterLinkColumnTemplate
    title="Credits"
  >
    <DesignCredit />
    <DeveloperCredit />
    <IconsCredit />
    <LogoDesignerCredit />
  </FooterLinkColumnTemplate>
);

const GitHubRepo = () => (
  <FooterLinkTemplate
    url="https://github.com/davidfeng88/bara"
    text="GitHub Repo"
  />
);

const PortfolioSite = () => (
  <FooterLinkTemplate
    url="https://davidfeng.us/"
    text="Portfolio Site"
  />
);

const GitHub = () => (
  <FooterLinkTemplate
    url="https://github.com/davidfeng88"
    text="GitHub"
  />
);

const LinkedIn = () => (
  <FooterLinkTemplate
    url="https://www.linkedin.com/in/gfeng/"
    text="LinkedIn"
  />
);

const AngelList = () => (
  <FooterLinkTemplate
    url="https://angel.co/ge-david-feng"
    text="AngelList"
  />
);

const DesignCredit = () => (
  <FooterLinkTemplate
    title="Design: "
    url="https://www.yelp.com/nyc"
    text="Yelp"
  />
);

const DeveloperCredit = () => (
  <FooterLinkTemplate
    title="Developer: "
    url="https://davidfeng.us/"
    text="Ge &#34;David&#34; Feng"
  />
);

const IconsCredit = () => (
  <FooterLinkTemplate
    title="Icons: "
    url="http://fontawesome.io/"
    text="Font Awesome"
  />
);

const LogoDesignerCredit = () => (
  <FooterLinkTemplate
    title="Logo Designer: "
    url="https://www.linkedin.com/in/meng-zhang-692b7644/"
    text="Meng Zhang"
  />
);

const FooterPicture = () => (
  <div className="footer-picture">
    <img alt="" src={window.staticImages.footerPic} />
  </div>
);

const FooterLinkColumnTemplate = ({
  title,
  children,
}) => (
  <div className="footer-links-col">
    <h3>{title}</h3>
    {children}
  </div>
);

FooterLinkColumnTemplate.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.node.isRequired,
};

const FooterLinkTemplate = ({
  title,
  url,
  text,
}) => (
  <p>
    {title}
    <a
      href={url}
      target="_blank"
      rel="noreferrer noopener"
    >
      {text}
    </a>
  </p>
);

FooterLinkTemplate.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string.isRequired,
  text: PropTypes.string.isRequired,
};

FooterLinkTemplate.defaultProps = {
  title: '',
  url: '',
  text: '',
};
