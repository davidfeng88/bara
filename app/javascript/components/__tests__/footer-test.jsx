import React from 'react';
import renderer from 'react-test-renderer';
import Footer from '../Footer';

window.staticImages = {
  footerPic: '',
};

it('renders Footer correctly', () => {
  const tree = renderer
    .create(<Footer />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
