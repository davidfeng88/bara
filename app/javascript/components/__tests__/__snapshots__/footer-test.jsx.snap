// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Footer correctly 1`] = `
<div
  className="footer"
>
  <div
    className="container footer-container"
  >
    <div
      className="footer-links"
    >
      <div
        className="footer-links-col"
      >
        <h3>
          About Bara
        </h3>
        <p>
          
          <a
            href="https://github.com/davidfeng88/bara"
            rel="noreferrer noopener"
            target="_blank"
          >
            GitHub Repo
          </a>
        </p>
      </div>
      <div
        className="footer-links-col"
      >
        <h3>
          About Me
        </h3>
        <p>
          
          <a
            href="https://davidfeng.us/"
            rel="noreferrer noopener"
            target="_blank"
          >
            Portfolio Site
          </a>
        </p>
        <p>
          
          <a
            href="https://github.com/davidfeng88"
            rel="noreferrer noopener"
            target="_blank"
          >
            GitHub
          </a>
        </p>
        <p>
          
          <a
            href="https://www.linkedin.com/in/gfeng/"
            rel="noreferrer noopener"
            target="_blank"
          >
            LinkedIn
          </a>
        </p>
        <p>
          
          <a
            href="https://angel.co/ge-david-feng"
            rel="noreferrer noopener"
            target="_blank"
          >
            AngelList
          </a>
        </p>
      </div>
      <div
        className="footer-links-col"
      >
        <h3>
          Credits
        </h3>
        <p>
          Design: 
          <a
            href="https://www.yelp.com/nyc"
            rel="noreferrer noopener"
            target="_blank"
          >
            Yelp
          </a>
        </p>
        <p>
          Developer: 
          <a
            href="https://davidfeng.us/"
            rel="noreferrer noopener"
            target="_blank"
          >
            Ge "David" Feng
          </a>
        </p>
        <p>
          Icons: 
          <a
            href="http://fontawesome.io/"
            rel="noreferrer noopener"
            target="_blank"
          >
            Font Awesome
          </a>
        </p>
        <p>
          Logo Designer: 
          <a
            href="https://www.linkedin.com/in/meng-zhang-692b7644/"
            rel="noreferrer noopener"
            target="_blank"
          >
            Meng Zhang
          </a>
        </p>
      </div>
    </div>
    <div
      className="footer-picture"
    >
      <img
        alt=""
        src=""
      />
    </div>
  </div>
</div>
`;
